src/Kafka/src/main/java/io/confluent/examples/clients/cloud/ConsumerExample.java

```diff
@@ -78,6 +85,11 @@ public static void main(final String[] args) throws Exception {
           System.out.printf("Consumed record with key %s and value %s, and updated total count to %d%n", key, value,
               total_count);
           // Make a reference to SocketIO -> Send data to connected clients
+          // This operation is called whe the event "dataevent" is called from the front end
```

Comment:
Not exactly true. This operation is called whenever a new record becomes available to the consumer. It then generates a new dataevent with the record as data. The front-end will then receive this dataevent and can use the record in the plots.